pkg/api/vm/handler.go

```diff
@@ -717,6 +719,43 @@ func (h *vmActionHandler) createVMImages(templateVersion *harvesterv1.VirtualMac
 	return nil
 }
 
+// Since the LH volume creation and replica scheduling are asynchronous
+// Even the CreateVolume csi call success, the replica scheduling may still fail
+// However for the sc with VolumeBindingMode is Immediate
+// we could check annotation for corresponding error if it exists
+func (h *vmActionHandler) checkForConventionSC(pvc *corev1.PersistentVolumeClaim) error {
+	sc, err := h.storageClassCache.Get(*pvc.Spec.StorageClassName)
+	if err != nil {
+		return err
+	}
+
+	// Even harvester specify default StorageClass,
+	// we check the sc existence for robustness
+	if sc == nil {
+		return nil
+	}
+
+	if *sc.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
+		return nil
+	}
+
+	if pvc.Status.Phase != corev1.ClaimBound {
+		return fmt.Errorf("volme %v not bound yet", pvc.Name)
+	}
+
+	pv, err := h.pvCache.Get(pvc.Spec.VolumeName)
+	if err != nil {
+		return err
+	}
+
+	scheduleErrAnno := pv.Annotations[longhorntypes.PVAnnotationLonghornVolumeSchedulingError]
+	if scheduleErrAnno == longhorn.ErrorReplicaScheduleInsufficientStorage {
+		return fmt.Errorf("volme %v with insufficient storage", pvc.Name)
```

Comment:
Since the annotation generally means that the PVC has an error condition, should we reject the attaching if we see any error here? More than just insufficient storage cases.

Reply:
Add rejecting volume attaching if [Longhorn reports any error](https://github.com/harvester/harvester/blob/117e04788b98d2778612c8181cecfcf776564e16/vendor/github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2/node.go#L36), except "replica scheduling failed" which means volume is in degrade mode but still can be attached.